src/NHapi.Base/Model/AbstractSegmentItem.cs

```diff
@@ -1,148 +1,148 @@
+        /// <param name="required">whether a value for this field is required in order for the segment
+        /// to be valid.</param>
+        /// <param name="maxReps">the maximum number of repetitions - 0 implies that there is no limit.</param>
+        /// <param name="length">the maximum length of each repetition of the field (in characters). </param>
+        /// <param name="constructorArgs">an array of objects that will be used as constructor arguments
+        /// if new instances of this class are created (use null for zero-arg constructor).</param>
+        /// <param name="description">Description of the segment.</param>
+        /// <throws>  HL7Exception if the given class does not inherit from IType or if it cannot be instantiated. </throws>
+        public AbstractSegmentItem(
+            Type t,
+            bool required,
+            int maxReps,
+            int length,
+            object[] constructorArgs,
+            string description)
+        {
+            if (!typeof(IType).IsAssignableFrom(t))
+            {
+                throw new HL7Exception(
+                    "Class " + t.FullName + " does not inherit from " + "NHapi.Base.Model.IType",
+                    ErrorCode.APPLICATION_INTERNAL_ERROR);
+            }
+
+            FieldType = t;
+            Length = length;
+            IsRequired = required;
+            this.maxReps = maxReps;
+            if (constructorArgs != null)
+            {
+                args.AddRange(constructorArgs);
+            }
+
+            Description = description;
+        }
+
+        /// <summary>
+        ///     The IType of this field in the segment.
+        /// </summary>
+        public Type FieldType { get; }
+
+        /// <summary>
+        /// Is this a required field.
+        /// </summary>
+        public bool IsRequired { get; }
+
+        /// <summary>
+        /// What is the length in characters of the field.
+        /// </summary>
+        public int Length { get; }
+
+        /// <summary>
+        /// Arguments to pass to a constructor for this field.
+        /// </summary>
+        public object[] Args
+        {
+            get { return args.ToArray(); }
+        }
+
+        /// <summary>
+        /// Maximum number of repetitions of this field.
+        /// </summary>
+        public int MaxRepetitions
+        {
+            get
+            {
+                if (maxReps <= 0)
+                {
+                    return int.MaxValue;
+                }
+                else
+                {
+                    return maxReps;
+                }
+            }
+
+            set
+            {
+                maxReps = value;
+            }
+        }
+
+        /// <summary>
+        /// What is this field.
+        /// </summary>
+        public string Description { get; set; }
+
+        /// <summary>
+        ///
+        /// </summary>
+        public IList<IType> Fields

```

Comment:
Can we get rid of the private field and just use the property?

Reply:
done